client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js

```diff
@@ -1,242 +1,249 @@
-
-    self.isSpecialNumber = false;
-    self.typologies = { france: [], belgium: [] };
-
-
-    // fetch list of billing accounts
-    return TelephonyMediator.getAll().then((groups) => {
-      self.billingAccounts = groups;
-      self.order.billingAccount = $stateParams.billingAccount;
-    }).catch((err) => {
-      TucToast.error(_.get(err, 'data.message'));
-      return $q.reject(err);
-    }).then(() => OvhApiOrder.v6().schema().$promise.then((schema) => {
-      if (schema && schema.models['telephony.NumberSpecialTypologyEnum'] && schema.models['telephony.NumberSpecialTypologyEnum'].enum) {
-        const typologies = _.map(schema.models['telephony.NumberSpecialTypologyEnum'].enum, typo => ({
-          value: typo,
-          label: $translate.instant(`telephony_order_specific_typology_${typo.replace(new RegExp('^be_|fr_'), '')}_label`),
+angular.module('managerApp')
+  .controller('TelecomTelephonyAliasPortabilityOrderCtrl',
+    function ($q, $scope, $stateParams, $translate, moment, TelephonyMediator,
+      OvhApiMe, OvhApiOrder, TucBankHolidays, TucToast, TELEPHONY_REPAYMENT_CONSUMPTION) {
+      const self = this;
+
+      // attributes shared by 'individual' and 'company' social reason
+      const sharedAttributes = [
+        'billingAccount', 'building', 'callNumber', 'city', 'comment', 'contactName',
+        'contactNumber', 'country', 'desireDate', 'displayUniversalDirectory', 'door',
+        'executeAsSoonAsPossible', 'firstName', 'floor', 'groupNumber', 'lineToRedirectAliasTo',
+        'name', 'offer', 'socialReason', 'stair', 'streetName', 'streetNumber', 'streetNumberExtra',
+        'streetType', 'type', 'zip',
+      ];
+
+      function init() {
+        self.order = {
+          // default values
+          executeAsSoonAsPossible: true,
+          type: 'landline',
+          sdatype: 'all',
+          socialReason: 'individual',
+          country: 'france',
+          contactNumber: '',
+          translatedCountry: $translate.instant('telephony_alias_portability_order_contact_country_france'),
+          displayUniversalDirectory: false,
+          numbersList: [],
+          success: false,
+          autoPay: false,
+          addressTooLong: false,
+        };
+
+        self.stepsList = ['number', 'contact', 'config', 'summary'];
+        self.step = 'number';
+        self.minDate = moment().add(15, 'days').toDate();
+        self.order.desireDate = moment(self.minDate).toDate();
+        self.desireDatePickerOpened = false;
+        self.isSDA = false;
+
+        // reset contract when step changes
+        $scope.$watch('PortabilityOrderCtrl.step', () => {
+          self.order.isContractsAccepted = false;
+        });
+
+        $scope.$watch('PortabilityOrderCtrl.order', () => {
+          self.order.contactName = self.order.name;
+        }, true);
+
+        $scope.$watch('PortabilityOrderCtrl.isSDA', () => {
+          self.order.offer = self.isSDA ? 'company' : 'individual';
+        }, true);
```

Comment:
I'm quite curious on why those are necessary and using a `ng-model` and listening to events such as `ng-change` like it is already partially done won't do 
https://github.com/ovh-ux/ovh-manager-telecom/blob/881f41fc439e5b2bea9f8280c8d7e38524a361d5/client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.html#L85

Reply:
I don't know why it was coded like that.